scripts/fu_storageutils.lua

```diff
@@ -6,23 +12,21 @@ function fu_storeItems(items, avoidSlots, spawnLeftovers)
 
 		for i = 0, world.containerSize(entity.id()) do -- iterate all output slots
 			local stack = world.containerItemAt(entity.id(), i) -- get the stack on i
-			if stack then -- not empty
-				if stack.name == something then -- its "something"
-					table.insert(slots,i) -- possible drop slot
-				end
+			if stack and (stack.name == something) then -- not empty and its "something"
+			  table.insert(slots,i) -- possible drop slot
```

Comment:
if condition1 then if condition2 then end end
will behave identically to 
if condition 1 and condition2 then end
because expressions on the right hand of 'and' will not be checked in the event the left hand is either false or nil.  'or' is the opposite case where in expressions to the right side of 'or' will not be evaluated unless the expression to the left is either 'false' or 'nil'

'If' will evaluate expressions in a boolean fashion and 'nil' is treated as 'not true' just like 'false' where as 'true' and any value that isn't 'false' or 'nil' will be treated as true for the purposes of boolean expression.

It really is equivalent behavior.  But take that part out if it really bothers you.